src/Time/Units.hs

```diff
@@ -120,6 +124,13 @@ type Fortnight   = 1209600 :% 1
 newtype Time (rat :: Rat) = Time { unTime :: RatioNat }
     deriving (Eq, Ord, Enum, Real, RealFrac, Generic)
 
+instance Semigroup (Time (rat :: Rat)) where
+    (<>) = (+)
+
+instance Monoid (Time (rat :: Rat)) where
+    mempty  = Time 0
+    mappend = (<>)
```

Comment:
We need to implement `sconcat` (from `Semigroup`) and `mconcat` (from `Monoid) using `foldl'` function because default implementations of these functions use `foldr` which makes them extremely inefficient.